lib/matplotlib/category.py

```diff
@@ -88,37 +87,41 @@ def __init__(self, seq):
 
 
 class UnitData(object):
-    # debatable makes sense to special code missing values
-    spdict = {'nan': -1.0, 'inf': -2.0, '-inf': -3.0}
-
     def __init__(self, data):
         """Create mapping between unique categorical values
         and numerical identifier
 
         Parameters
         ----------
         data: iterable
-            sequence of values
+              sequence of values
```

Comment:
The sequence can contain anything including mixed types right? Might be worth adding a note here if that's true.